### PR TITLE
Added a new s6-based image for Consul.

### PR DESCRIPTION
**Pull request by smebberson**
@sandytrinh, this is ready for review.

The README.md has the information on how to use the image. This doesn't have support for running Consul in multiple modes at this point, as we discussed. But it has full support for overriding the default configuration.

### CODE REVIEW

**Comment by smebberson** [comment]
@sandytrinh, this needs some updates. I want to make it a little more generic.

**Comment by smebberson** [comment]
@sandytrinh, I've made some improvements. Please continue reviewing.

File: consul/README.md [hunk 1]
```diff
@@ -0,0 +1,63 @@
+Consul
+======
+
+An image for using [consul][Consul] within Docker containers.
+
+**_Yet another container for running MongoDB?_**
```
Comment (sandytrinh) [inline consul/README.md]: Please change `MongoDB` to `consul`.

**Comment by smebberson** [comment]
@sandytrinh, this is ready for review again. `consul` is now running under the unprivileged user `consul`.

File: consul/root/etc/consul.d/bootstrap/config.json [hunk 2]
```diff
@@ -0,0 +1,6 @@
+{
+    "bootstrap_expect": 1,
+    "server": true,
+    "node_name": "agent-bootstrapped",
+    "data_dir": "/data"
```
Comment (sandytrinh) [inline consul/root/etc/consul.d/bootstrap/config.json:5]: Do we need `client_addr` setting here so that we can easily join consul servers?

Comment (sandytrinh) [inline consul/root/etc/consul.d/bootstrap/config.json:5]: This is not required, please ignore!

**Comment by sandytrinh** [comment]
@smebberson This is working great, please merge!

**Pull request merged**